modules/network/multivpc/main.tf

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+locals {
+  # this input variable is validated to be in CIDR format
+  super_global_ip_cidr_prefix = split("/", var.super_global_ip_address_range)[1]
+  network_new_bits            = var.network_cidr_prefix - local.super_global_ip_cidr_prefix
+  subnetwork_bits             = ceil(log(var.network_count, 2))
+  additional_networks = [
+    for vpc in module.vpcs :
+    {
+      network            = null
+      subnetwork         = vpc.subnetwork_name
+      subnetwork_project = var.project_id
+      network_ip         = ""
+      nic_type           = "GVNIC"
```

Comment:
Looking at this, I think we should parameterize network outputs (e.g. default to GVNIC, but not force user to use GVNIC). We can discuss offline.

Reply:
I think this is accomplished with the new `network_template`